src/Graphics/SpriteBatch.cs

```diff
@@ -1148,11 +1107,146 @@ byte effects
 				}
 				DrawPrimitives(texture, 0, 1);
 			}
-			else
+			else if (sortMode == SpriteSortMode.Deferred)
 			{
+				// We won't be sorting so go ahead load straight into the vertex info buffer.
+				fixed (VertexPositionColorTexture4* sprite = &vertexInfo[numSprites])
+				{
+					UpdateVertexInfo(
+						sprite,
+						sourceX,
+						sourceY,
+						sourceW,
+						sourceH,
+						destinationX,
+						destinationY,
+						destinationW,
+						destinationH,
+						color,
+						originX,
+						originY,
+						rotationSin,
+						rotationCos,
+						depth,
+						effects);
+				}
+
 				textureInfo[numSprites] = texture;
+
 				numSprites += 1;
 			}
+			else
+			{
+				// We will be sorting, so queue up the information to sort by reference.
+				SpriteInfo spriteInfo = spriteQueue[numSprites];
+				spriteInfo.texture = texture;
+				spriteInfo.textureHash = texture.GetHashCode();
+				spriteInfo.sourceX = sourceX;
+				spriteInfo.sourceY = sourceY;
+				spriteInfo.sourceW = sourceW;
+				spriteInfo.sourceH = sourceH;
+				spriteInfo.destinationX = destinationX;
+				spriteInfo.destinationY = destinationY;
+				spriteInfo.destinationW = destinationW;
+				spriteInfo.destinationH = destinationH;
+				spriteInfo.color = color;
+				spriteInfo.originX = originX;
+				spriteInfo.originY = originY;
+				spriteInfo.rotationSin = rotationSin;
+				spriteInfo.rotationCos = rotationCos;
+				spriteInfo.depth = depth;
+				spriteInfo.effects = effects;
+
+				numSprites += 1;
+			}
+		}
+
+		[MethodImpl(256)]
```

Comment:
This is a "hack" to allow the JIT to inline methods larger than 32 IL instructions. It is equivalent to [MethodImpl(MethodImplOptions.AggressiveInlining)] but the enum flag MethodImplOptions.AggressiveInlining = 256 is not available in the version of .NET Framework FNA is using right now. It's supposed to work, but I haven't measured it's impact yet.  I will do the measurements and post if it makes any difference.

Reply:
Makes sense - we'll probably have to cite this, to avoid it looking like a magic number: 

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.methodimploptions?view=netframework-4.7.2

And it this does work, might be worth looking at for #227 too.

Reply:
I don't see a measurable difference with it.  I tried it with, without, and with the NoInline attribute.  I'll need to try it out in isolation I think.  I can just take it out.